### ISSUE TITLE
Generic training knobs exist only on the research fixtures: `toy train from-scratch --lr` is rejected

### ISSUE THREAD

**Issue by OriPekelman** [OriPekelman/toy#174]
Found while drawing the capability/fixture seam. One line states it:

```
--lr  is accepted by  franken franken-moe mlp ctr gnn ssm lstm gtx diff ae difflm
--lr  is REJECTED by  from-scratch          <- the README's own quickstart recipe
```

`toy train from-scratch` could not set a learning rate, model width, depth, context or vocabulary — six flags total — while `franken-moe` had forty-five including a five-knob adaptive LR controller.

That is not a missing feature so much as a mis-wiring. Generic capability got built **into the fixtures** as each research question needed it, instead of into the framework, because the fixtures were where the work was happening. The result is a framework whose product path is less capable than its test lanes, and a user whose only route to a learning rate is to pick a synthetic research lane.

## Scope

Promote onto `from-scratch` / `warm-start`: `--lr`, `--d-model`, `--layers`, `--heads`, `--d-ff`, `--context`, `--vocab`, and `--warmup` (`warm-start` only).

Non-negotiable: **every default is the existing gate-fixed value**, so an invocation setting none of them stays byte-identical and every recorded run stays reproducible. Gated against `prep/fixtures/train_baseline.txt` and `train_warm_start_baseline.txt`.

## Explicitly out of scope, and why

`--batch` and `--val-batches` do **not** get promoted. `from-scratch` trains on the first line of `data/ts_seqs.txt` repeated `--steps` times: no batch dimension, no eval loop. The flags would parse and do nothing. Making them real is a data-pipeline feature, tracked client-side as tao#28.

Same reasoning scopes `--warmup` to `warm-start`: `from-scratch` uses a constant hyper-parameter and has no schedule, so the flag is **rejected** there rather than silently ignored.

## Two latent defects in the same code

1. **The reported learning rate is a literal.** `config.add_raw("lr", "0.001")` in `run_start` and again per step — a hard-coded string, not the optimizer's value. Correct today by coincidence; a change to `AdamW.for_from_scratch` would have made every bundle record a rate the run did not use, silently. Same "listed in one place, applied from another" shape as toy#171 (prereq drift), the ggml patch dropped by every fresh clone, and the ungeneratable CUDA mirrors.

2. **`--vocab` can index out of bounds.** The label builder is the *unguarded* one, justified by a comment about a known-good first line — true while the vocabulary was a literal, false once it is settable. Needs a fail-loud check against the corpus's highest token id.

Client notice: tao#28.

**Comment by OriPekelman** [comment]
Shipped in 713285b. Battery green (57 legs).

All eight knobs promoted onto `from-scratch`/`warm-start`, byte-null by default and verified through the CLI against both stored baselines.

Both latent defects fixed: the reported learning rate is now derived from the optimizer instead of being a hard-coded `"0.001"`, and `--vocab` below the corpus's highest token id fails loud (boundary verified exact — 26 fails, 27 runs).

`--batch`/`--val-batches` stayed out as scoped, and `docs/cli.md` explains why rather than leaving the gap to read as an oversight. Client notice: tao#28.
